src/modules/quarto.rs

```diff
@@ -0,0 +1,89 @@
+use super::{Context, Module, ModuleConfig};
+
+use crate::configs::quarto::QuartoConfig;
+use crate::formatter::{StringFormatter, VersionFormatter};
+
+/// Creates a module with the current Quarto version
+pub fn module<'a>(context: &'a Context) -> Option<Module<'a>> {
+    let mut module = context.new_module("quarto");
+    let config = QuartoConfig::try_load(module.config);
+
+    let is_quarto_project = context
+        .try_begin_scan()?
+        .set_files(&config.detect_files)
+        .set_extensions(&config.detect_extensions)
+        .set_folders(&config.detect_folders)
+        .is_match();
+
+    if !is_quarto_project {
+        return None;
+    }
+
+    let parsed = StringFormatter::new(config.format).and_then(|formatter| {
+        formatter
+            .map_meta(|var, _| match var {
+                "symbol" => Some(config.symbol),
+                _ => None,
+            })
+            .map_style(|variable| match variable {
+                "style" => Some(Ok(config.style)),
+                _ => None,
+            })
+            .map(|variable| match variable {
+                "version" => {
+                    let version = context.exec_cmd("quarto", &["--version"])?.stdout;
```

Comment:
Please handle the trailing newline and update the command mocks to add it.

Reply:
Thank you, done.